packages/keyboard-shortcuts/src/hooks/use-shortcut.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * External dependencies
+ */
+import Mousetrap from 'mousetrap';
+import 'mousetrap/plugins/global-bind/mousetrap-global-bind';
+import { includes } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { rawShortcut } from '@wordpress/keycodes';
+import { useEffect } from '@wordpress/element';
+
+/**
+ * Return true if platform is MacOS.
+ *
+ * @param {Object} _window   window object by default; used for DI testing.
+ *
+ * @return {boolean}         True if MacOS; false otherwise.
+ */
+function isAppleOS( _window = window ) {
+	const { platform } = _window.navigator;
+
+	return platform.indexOf( 'Mac' ) !== -1 ||
+		includes( [ 'iPad', 'iPhone' ], platform );
+}
+
+/**
+ * Attach a keyboard shortcut handler.
+ *
+ * @param {string} name       Shortcut name.
+ * @param {Function} callback Shortcut callback.
+ * @param {Object} options    Shortcut options.
+ */
+function useShortcut( name, callback, {
+	bindGlobal = false,
+	eventName = 'keydown',
+	target,
+} ) {
+	const { combination, aliases } = useSelect( ( select ) => {
+		return {
+			combination: select( 'core/keyboard-shortcuts' ).getShortcutKeysCombination( name ),
+			aliases: select( 'core/keyboard-shortcuts' ).getShortcutAliases( name ),
+		};
+	}, [ name ] );
+
+	useEffect( () => {
+		const shortcuts = [ combination, ...aliases ].filter( ( shortcut ) => !! shortcut );
+		const mousetrap = new Mousetrap( target ? target.current : document );
+		const shortcutKeys = shortcuts.map( ( shortcut ) => {
+			return shortcut.modifier ?
+				rawShortcut[ shortcut.modifier ]( shortcut.character ) :
+				shortcut.character;
+		} );
+
+		shortcutKeys.forEach( ( shortcut ) => {
+			const keys = shortcut.split( '+' );
+			const modifiers = new Set( keys.filter( ( value ) => value.length > 1 ) );
```

Comment:
What's the single-character modifier we're filtering here? Comment or externalized/named function could help clarify.

Reply:
I can't tell. This is ported from KeyboardShortcuts.

Reply:
It looks like it determines whether a key is a modifier by the length of the string. E.g. if I add a pass a shortcut `Shift+Cmd+M`, it'll determine that the modifiers are Shift and Cmd because they're not a single character.

That does seem a little bit unusual.

Reply:
> It looks like it determines whether a key is a modifier by the length of the string. E.g. if I add a pass a shortcut `Shift+Cmd+M`, it'll determine that the modifiers are Shift and Cmd because they're not a single character.

That sounds about right.
 
>Might be safer to check against an array of known modifiers.

I'd have been content if the original implementation had simply added an inline comment, to save ourselves the struggle in trying to decipher its meaning 😅